core/src/main/java/io/neow3j/transaction/Transaction.java

```diff
@@ -169,18 +169,14 @@ public String getTxId() throws IOException {
      * @return the Neo node's response.
      * @throws IOException                       if a problem in communicating with the Neo node
      *                                           occurs.
-     * @throws TransactionConfigurationException if signatures are missing for one or more signers
-     *                                           of the transaction.
+     * @throws TransactionConfigurationException if the number of signers and witnesses on the
+     *                                           transaction are not equal.
      */
     public NeoSendRawTransaction send() throws IOException {
-        List<ScriptHash> witnesses = this.getWitnesses().stream()
-                .map(Witness::getScriptHash).collect(Collectors.toList());
-
-        for (Signer signer : this.getSigners()) {
-            if (!witnesses.contains(signer.getScriptHash())) {
-                throw new TransactionConfigurationException("The transaction does not have a "
-                        + "signature for each of its signers.");
-            }
+        if (getSigners().size() != getWitnesses().size()) {
+            throw new TransactionConfigurationException("The transaction does not have the same " +
+                    "number of signers and witnesses. For every signer there has to be one " +
+                    "witness, even if that witness is empty.");
```

Comment:
Important change to review.

Reply:
Yes. Good. This change is done since now we need to add "empty" witnesses.

One use case we could test in the future:
Is there a use case that two empty witnesses are added (like... two signers without priv keys)? If so, the Neo node would accept that? Or both would feed to be translated into a SINGLE empty witness?

This is not required for now, but it would be good to know....

Reply:
If a transaction transfers tokens from two different smart contracts that would lead to exactly the scenario you described. The two contracts would be added as signers and two empty witnesses have to be added (and probably a signature from the contract owner, depending on the contracts' `verify` methods).

Reply:
@csmuller should we create a GitHub issue for this one?
Also, about investigating whether combining multiple empty Witnesses into one is the right way of doing things or not.

If you open an issue about that, then we could merge this PR.

Reply:
Alright, I created issue #431 for this problem.